src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -695,12 +694,18 @@ private void setBlockFromInventory(@NotNull final BlockPos location, final Block
         {
             slot = worker.findFirstSlotInInventoryWith(block, block.getMetaFromState(metadata));
         }
+
         if (slot != -1)
         {
-            getInventory().decrStackSize(slot, 1);
-            //Flag 1+2 is needed for updates
-            world.setBlockState(location, metadata, 3);
+            if (getInventory().extractItem(slot, 1, false) != null)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)